orangecontrib/prototypes/widgets/owexplpred.py

```diff
@@ -0,0 +1,409 @@
+    minIter : int
+        minimum number of iterations per attiribute
+    seed : int
+        seed for the numpy.random generator, default is 0
+
+    Returns:
+    -------
+    classValue: float
+        either index of predicted class or predicted value
+    table: Orange.data.Table
+        table containing atributes and corresponding contributions, descending
+
+    """
+
+    def __init__(self, data, model, pError=0.05, error=0.05, batchSize=100, maxIter=59000, minIter=100, seed=0):
+
+        self.model = model
+        self.data = data
+        self.pError = pError
+        self.error = error
+        self.batchSize = batchSize
+        self.maxIter = maxIter
+        self.minIter = minIter
+        self.atr_names = DiscreteVariable(name='attributes',
+                                          values=[var.name for var in data.domain.attributes])
+        np.random.seed(seed)
+
+    def anytime_explain(self, instance, callback=None):
+        dataRows, noAtr = self.data.X.shape
+        classValue = self.model(instance)[0]
+
+        # placeholders
+        steps = np.zeros((1, noAtr), dtype=float)
+        mu = np.zeros((1, noAtr), dtype=float)
+        M2 = np.zeros((1, noAtr), dtype=float)
+        expl = np.zeros((1, noAtr), dtype=float)
+        var = np.ones((1, noAtr), dtype=float)
+
+        atr_indices = np.asarray(range(noAtr)).reshape((1, noAtr))
+        batchMxSize = self.batchSize * noAtr
+        zSq = abs(st.norm.ppf(self.pError/2))**2
+
+        tiled_inst = Table.from_numpy(instance.domain,
+                                      np.tile(instance.X, (self.batchSize, 1)), np.full((self.batchSize, 1), instance.Y[0]))
+        inst1 = copy.deepcopy(tiled_inst)
+        inst2 = copy.deepcopy(tiled_inst)
+        iterations_reached = np.zeros((1, noAtr))
+
+        while not(all(iterations_reached[0, :] > self.maxIter)):
+            if not(any(iterations_reached[0, :] > self.maxIter)):
+                a = np.random.choice(atr_indices[0], p=(
+                    var[0, :]/(np.sum(var[0, :]))))
+            else:
+                a = np.argmin(iterations_reached[0, :])
+
+            perm = np.random.choice([True, False], batchMxSize, replace=True)
+            perm = np.reshape(perm, (self.batchSize, noAtr))
+            rand_data = self.data.X[np.random.randint(
+                dataRows, size=self.batchSize), :]
+            inst1.X = np.copy(tiled_inst.X)
+            inst1.X[perm] = rand_data[perm]
+            inst2.X = np.copy(inst1.X)
+
+            inst1.X[:, a] = tiled_inst.X[:, a]
+            inst2.X[:, a] = rand_data[:, a]
+            f1 = self._get_predictions(inst1, classValue)
+            f2 = self._get_predictions(inst2, classValue)
+
+            diff = np.sum(f1 - f2)
+            expl[0, a] += diff
+
+            # update variance
+            steps[0, a] += self.batchSize
+            iterations_reached[0, a] += self.batchSize
+            d = diff - mu[0, a]
+            mu[0, a] += d/steps[0, a]
```

Comment:
Try to be consistent in the use of spaces around binary operators. Unless there are multiple operators with different priorities, always using spaces should be fine.
https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements